funding/manager.go

```diff
@@ -3488,6 +3524,19 @@ func (f *Manager) newChanAnnouncement(localPubKey,
 	// max_htlc field.
 	msgFlags := lnwire.ChanUpdateOptionMaxHtlc
 
+	// Assume default fee parameters defined in the config
+	baseFee := uint32(f.cfg.DefaultRoutingPolicy.BaseFee)
+	feeRate := uint32(f.cfg.DefaultRoutingPolicy.FeeRate)
+
+	// Check if we should use the channel fees provided
+	// in the initFundingMsg.
+	if forwardingPolicy, ok := f.channelFeeMap[chanID]; ok {
```

Comment:
we could make this the default flow and ensure that all pending channels are always in the map. If they did not have custom fees they are added to the map with the default routing police values. However, we need to be sure that the map gets populated during the lnd starting processes

Reply:
Hmm, what if a new entry point to this flow is added but the `f.channelFeeMap` is not initialized. How does the implementer of the new flow know that they need to add to the map? Since we can't predict who is calling `newChanAnnouncement` and how they are calling it in the future I would keep the default fees here. Unless you insist :-).

Reply:
I would be consistent. As a user, if at some point I opened a channel with custom fees and I see it propagated with the default ones I won't be happy with "well, it is just that it followed another flow and the announcer did not have the data with your custom fees".

I think it is better if we return an error when trying to announce a channel if it is not present in `channelFeeMap`.